backend.native/cli.bc/src/org/jetbrains/kotlin/cli/bc/K2NativeCompilerArguments.kt

```diff
@@ -157,6 +157,14 @@ class K2NativeCompilerArguments : CommonCompilerArguments() {
     @Argument(value = "-Xruntime", deprecatedName = "--runtime", valueDescription = "<path>", description = "Override standard 'runtime.bc' location")
     var runtimeFile: String? = null
 
+    @Argument(
+        value = SOURCE_LIBRARY_ARG,
+        valueDescription = "<path>",
+        description = "A library to be processed in the same manner as source files.\n" +
+                "Must be one of libraries passed with '-library'"
+    )
+    var sourceLibraries: Array<String>? = null
```

Comment:
A couple of trivial observations:
* Exported libraries and source libraries are somewhat similar.
* Source libraries and sources are quite similar (in other way).

Similar things should be handled similarly. I believe we should consider making some conclusions from this (maybe later).

For example:
1. A source library should be passed only once (e.g. should not require passing it with `-l` too).
2. A source library should be passed with its path specified (i.e. with no repo-based resolve involved).

Not sure if '1' is applicable to exported libraries though.
Btw, is "source library" a library at all? I would rather say that "source library" is much closed to "source" than to "library" (including an exported one).

Reply:
> Btw, is "source library" a library at all?

It looks like a mental model question. Currently I prefer consider source libraries as something similar to exported and covered libraries. I.e. they _are_ _libraries_ but they also require some additional processing from the compiler side. The main difference is that for source libraries this processing definition is less formal ("processed as sources" is more fuzzy than "analyzed by code coverage"). But of course this mental model can be reconsidered. May be invite @alexander-gorshenev and discuss it offline?

> Not sure if '1' is applicable to exported libraries though.

What cons do you see here?